src/app-layer-ssl.c

```diff
@@ -143,11 +146,101 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            input += SSLV3_CLIENT_HELLO_VERSION_LEN;
+
+            /* skip random */
+            input += SSLV3_CLIENT_HELLO_RANDOM_LEN;
+
+            if ((input - initial_input) > input_len)
+                goto end;
+
+            /* skip session id */
+            uint8_t session_id_length = *(input++);
+
+            input += session_id_length;
+
+            if ((input + 1 - initial_input) > input_len)
```

Comment:
That would make the code look nicer :)

Do you mean a macro looking something like this?

```c
#define HAS_SPACE(n) (((input) + (n) - (initial_input)) >= (input_len)) ?  0 : 1
```